application/controllers/admin/useraction.php

```diff
@@ -670,6 +667,25 @@ public function personalsettings()
                 }
             }
 
+            if (Yii::app()->request->getPost('newemailshown') == "1") {
+                if (Yii::app()->getConfig('demoMode')) {
+                    Yii::app()->setFlashMessage(gT("You can't change email if demo mode is active."), 'error');
+                    $this->getController()->redirect(array("admin/user/sa/personalsettings"));
```

Comment:
My opinion : it's better to not redirect NOW, but allow other update (else all are lost, not good for web quality) .
Best is to disable saveandclose
https://github.com/LimeSurvey/LimeSurvey/blob/41780805861f2dd88bf5513d9d12583593aeac02/application/controllers/admin/useraction.php#L725

something like 
`if($noerror && Yii::app()->request->getPost("saveandclose")) {`

Reply:
> It's better to not redirect NOW, but allow other update (else all are lost, not good for web quality) .

This would be complex as then 2 alerts would appear. One saying "no passowrd changes are allowed on demo" and another one saying "some changes couldn't be done".

What we could do is to disable the ability to change some fields if on demo mode.
That would be the easiest.

Reply:
> This would be complex as then 2 alerts would appear. One saying "no password changes are allowed on demo" and another one saying "some changes couldn't be done".

No : if you disable other save : user lost is update and did not know what is updated or not …

And her e: i put example on demo mode, but there are 2 point

1. Use password only to save new password and email, not the other things 
2. Then you must disallow update email (and show an error) but save other part 
3. If there are an error somewhere (demo, password, and anything else) : you must get back to the form.